openmc/material.py

```diff
@@ -795,16 +795,31 @@ def get_elements(self):
 
         return sorted({re.split(r'(\d+)', i)[0] for i in self.get_nuclides()})
 
-    def get_nuclides(self):
-        """Returns all nuclides in the material
+    def get_nuclides(self, element: Optional[str] = None):
+        """Returns a sorted list of all nuclides in the material, if the
+        argument element is specified then just nuclides of that element are
+        returned.
+
+        Parameters
+        ----------
+        element : str
+            Specifies the element to match when searching through the nuclides
 
         Returns
         -------
         nuclides : list of str
             List of nuclide names
 
         """
-        return [x.name for x in self._nuclides]
+        if element:
+            matching_nuclides = []
+            for nuclide in self._nuclides:
+                if re.split(r'(\d+)', nuclide.name)[0] == element:
+                    matching_nuclides.append(nuclide.name)
+            return sorted(set(matching_nuclides))
+
+        else:
+            return sorted(set([x.name for x in self._nuclides]))
```

Comment:
I don't think we should change the order that nuclides are returned in. It seems most natural to return nuclides in the order that they were originally specified.

Reply:
Yes, agreed. I will close this PR and make a clean one without the sorting shortly.